packages/protocol/contracts/identity/Random.sol

```diff
@@ -48,23 +49,35 @@ contract Random is IRandom, Ownable, Initializable {
    * @param randomness Bytes that will be added to the entropy pool.
    * @param newCommitment The hash of randomness that will be revealed in the future.
    * @param proposer Address of the block proposer.
+   * @param isLastEpochBlock If current block is the last of its epoch.
    * @dev If the Random contract is pointed to by the Registry, the first transaction in a block
    * should be a special transaction to address 0x0 with 64 bytes of data - the concatenated
    * `randomness` and `newCommitment`. Before running regular transactions, this function should be
    * called.
    */
-  function revealAndCommit(bytes32 randomness, bytes32 newCommitment, address proposer) external {
+  function revealAndCommit(
+    bytes32 randomness,
+    bytes32 newCommitment,
+    address proposer,
+    bool isLastEpochBlock
```

Comment:
Rather than have this passed as a boolean can we determine this from `block.number`? UsingPrecompiles should have some functions that can help